ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/IdentityProviderAwareAuthServiceImpl.scala

```diff
@@ -90,16 +91,29 @@ class IdentityProviderAwareAuthServiceImpl(
         Future.successful(payload)
     }
 
-  private def parse(jwtPayload: String): AuthServiceJWTPayload = {
+  private def parse(jwtPayload: String): AuthServiceJWTPayload = if (expectsAudienceBasedTokens)
+    parseAudienceBasedPayload(jwtPayload)
```

Comment:
Do I understand this correctly. When participant is set up for audience based tokens, that's all that's accepted?

Reply:
yes and I don't see the reason of why not

Reply:
This also gives us a decent forward migration story for actually getting rid of the legacy tokens at some point.